### PR TITLE
Automatically save the RX initial rate on disconnect

### PR DESCRIPTION
**Pull request by CapnBry**
Automatically save the RX initial rate on disconnect! Did I stutter?!

We see this question a lot about how slowly ExpressLRS connects (or _binds_ 😡) and nobody knows about the setting to make it start faster. This PR just saves current packet rate when the receiver disconnects. This occurs every time the packet rate is changed via the Lua too, since that's a forced disconnect. I do not do it on connect, as that could cause the RX to stall during the flash write and lose connection.

I initially wrote InitRate as just a config item due to the danger of trying to do an flash write operation while flying and you just failsafed-- it is like the one time you need your receiver to be back up and running as fast as possible. However, I've been flying with this code for months now, with plenty of failsafe testing, and haven't experienced unexpected behavior.

### Unhappy case
In the case where:
* The receiver has a different InitRate than the handset, e.g. a new receiver on an existing EdgeTX model
* The receiver eventually connects, pilot flies, wheee! ✈
* The pilot does the right thing and unplugs the battery before turning off the handset
* RESULT: InitRate is not updated, next boot's connect takes just as long :(

Workaround: Turn off the handset while connected (haha what could go wrong), or set the packet rate again while the receiver is connected. 

## VOTE!
👍 we remove the Lua config item too
👎 leave the RX Lua "Init Rate" item even though we do not respect the user's choice and overwrite it all the time. This adds another workaround to the unhappy case, using the Lua to set it. However, I think people will get upset and confused if we leave it in how they "always change the InitRate and it **always** keeps resetting to some random value"

### CODE REVIEW

**Comment by pkendall64** [comment]
Why not change the config on connection rather than disconnection?

**Comment by deadbytefpv** [comment]
I'm fine removing it from the Lua, as long as we let users know why it was removed and how it works now, perhaps via the Release Notes that seemingly no one reads. 

Kinda holding off "How to use the Lua" page update with the new options.

**Comment by P-I-Engineer** [comment]
i just helped people on the book of faces with this. Glad to see you're always ahead of the game with solutions.

**Comment by CapnBry** [comment]
> I'm fine removing it from the Lua, as long as we let users know why it was removed and how it works now, perhaps via the Release Notes that seemingly no one reads.

Yeah that is a really good point. It is a simple "the receiver will start at the last rate it was at on the last time it disconnected". Now that I type that, it really is clunky, but I will workshop it a bit for the docs website. 

It can be removed from the lua section (is this our first lua removal?! We need an obsolete lua items page now too!) Where do you think would be the best pllace for me to put a paragraph describing the feature?

**Comment by deadbytefpv** [comment]
> > I'm fine removing it from the Lua, as long as we let users know why it was removed and how it works now, perhaps via the Release Notes that seemingly no one reads.
> 
> Yeah that is a really good point. It is a simple "the receiver will start at the rate it was on the last time it disconnected". Now that I type that, it really is clunky, but I will workshop it a bit for the docs website. Just need to be clear the receiver has to disconnect, not just unplugged while connected.
> 
> It can be removed from the lua section (is this our first lua removal?! We need an obsolete lua items page now too!) Where do you think would be the best pllace for me to put a paragraph describing the feature?

I'm doing an update on the Using the Lua Script page to finally include Other Options. I will put Init Rate there and we can simply add a Deprecated/Removed notice.

**Comment by mha1** [comment]
We tried this PR and coincidentally found a problem using a Gemini receiver with a non-Gemini module. Telemetry is not working in this case. Setting the initial rate for the next connect for RX targets only seems to solve the issue. 

```
void LostConnection(bool resumeRx)
{
    DBGLN("lost conn fc=%d fo=%d", FreqCorrection, hwTimer::getFreqOffset());

#if defined(TARGET_RX)
    // Use this rate as the initial rate next time if we connected on it
    if (connectionState == connected)
        config.SetRateInitialIdx(ExpressLRS_nextAirRateIndex);
#endif
...
```

**Comment by CapnBry** [comment]
Hey great, @mha1 but can you clarify? rx_main.cpp should only be included in `TARGET_RX` and how would this have any effect on telemetry at all? Can you give a detailed explanation:
* The error
* RX and TX hardware
* Target you're building and Unified config name

**Comment by mha1** [comment]
This is kinda embarrassing. You are of course absolutely right. Forget about this (I deleted the comment) and let me try again from scratch.

**Comment by mha1** [comment]
Built everything from scratch again. No issues. My sincerest apologies. The only excuse I have is shitty LBT. Most likely rx and module to close to each other. Had that before and hadn't remembered.

**Comment by CapnBry** [comment]
> Built everything from scratch again. No issues.

No worries. You had me in a panic trying to figure out _hoowwwwwwww_ and then how the change you posted could fix that. Glad it wasn't the issue.

**Comment by mha1** [comment]
Send me your PayPal, I owe you a beer.

**Comment by aowi7280** [comment]
So what is "the setting to make it start faster"?

**Comment by CapnBry** [comment]
> So what is "the setting to make it start faster"?

The "Init Rate" setting in the RX's Lua that this would replace.

**Comment by schugabe** [comment]
We could show the status of the initial config in the lua settings of the rx so the user could check the settings if startup is slow? Or maybe on the config web page if we don't want to invest the memory :-D

**Comment by CapnBry** [comment]
haha oh boy, I feel like if it was just a display item people would contact support wondering why they can't change it and then we're back to square one. Note that the web page still takes memory to be able to build the json, and ESP doesn't have any RAM to spare (especially when running in AP mode) so I think we should save the RAM for useful things and not just for display stuff that would not be used.

It's internal. If the receiver takes a long time to connect, then just wait for it to connect, turn off the handset. When the LED on the RX starts blinking again, it is saved and next time it will be fast.

**Review by schugabe — APPROVED** [review]

**Review by pkendall64 — APPROVED** [review]
